core/src/main/java/com/linecorp/armeria/client/ArrayBasedEventLoopState.java

```diff
@@ -0,0 +1,150 @@
+                             DefaultEventLoopScheduler scheduler) {
+        super(eventLoops, scheduler);
+        this.maxNumEventLoops = maxNumEventLoops;
+        entries = new AbstractEventLoopEntry[maxNumEventLoops];
+        if (eventLoops.size() == maxNumEventLoops) {
+            init(0);
+        } else {
+            init(scheduler().acquisitionStartIndex(maxNumEventLoops));
+        }
+    }
+
+    private void init(final int acquisitionStartIndex) {
+        final int initialEventLoopOffset = ThreadLocalRandom.current().nextInt(maxNumEventLoops);
+        for (int i = 0; i < maxNumEventLoops; ++i) {
+            final int nextIndex = (acquisitionStartIndex + (initialEventLoopOffset + i) % maxNumEventLoops) %
+                                  eventLoops().size();
+            entries[i] = new Entry(this, eventLoops().get(nextIndex), i);
+        }
+    }
+
+    private AbstractEventLoopEntry targetEntry() {
+        int minActiveRequest = Integer.MAX_VALUE;
+        int targetIndex = 0;
+        for (int i = 0; i < maxNumEventLoops; ++i) {
+            final AbstractEventLoopEntry e = entries[i];
+            final int activeRequests = e.activeRequests();
+            if (activeRequests == 0) {
+                return e;
+            }
+            if (minActiveRequest > activeRequests) {
+                minActiveRequest = activeRequests;
+                targetIndex = i;
+            }
+        }
+        return entries[targetIndex];
+    }
+
+    @Override
+    AbstractEventLoopEntry acquire() {
+        lock();
+        try {
+            final AbstractEventLoopEntry e = targetEntry();
+            e.incrementActiveRequests();
+            allActiveRequests++;
+            return e;
+        } finally {
+            unlock();
+        }
+    }
+
+    @Override
+    void release(AbstractEventLoopEntry e) {
+        lock();
+        try {
+            e.decrementActiveRequests();
+            if (--allActiveRequests == 0) {
+                setLastActivityTimeNanos();
+            }
+        } finally {
+            unlock();
+        }
+    }
+
+    @Override
+    AbstractEventLoopEntry[] entries() {
+        return entries;
+    }
+
+    @Override
+    int allActiveRequests() {
+        return allActiveRequests;
+    }
+
+    private static final class Entry extends AbstractEventLoopEntry {
+
+        private final int id;
+
+        private int activeRequests;
+
+        Entry(AbstractEventLoopState parent, EventLoop eventLoop, int id) {
+            super(parent, eventLoop);
+            this.id = id;
+        }
+
+        @Override
+        int activeRequests() {
+            return activeRequests;
+        }
+
+        @Override
+        void incrementActiveRequests() {
+            activeRequests++;
+        }
+
+        @Override
+        void decrementActiveRequests() {
+            activeRequests--;
+        }
+
+        @Override
+        int id() {
+            return id;
+        }
```

Comment:
This method is not used for `ArrayBasedEventLoopState`.

Reply:
However, some tests use this method, so it should be implemented.